package.json

```diff
@@ -27,13 +27,16 @@
     "telegraf": "./bin/telegraf"
   },
   "scripts": {
-    "lint": "eslint src/",
-    "pretest": "tsc",
+    "typecheck": "tsc --noEmit",
+    "build": "npm run build:lib && npm run build:docs",
+    "build:lib": "npm run prepare",
+    "build:docs": "tsc --build docs/examples && npm run typedoc",
```

Comment:
It is very convenient to have those shortcuts here. If we should try not to use `package.json` as a build script container, we may want to have some kind of alternative that supports a similarly easy way to build the project.

Reply:
I understand `build`, but what's the advantage of `build:lib` over `prepare` and `build:docs` over `typedoc`?

Reply:
I'd be fine with renaming `typedoc` to `build:docs` actually. `prepare` could be changed to `tsc --build docs/examples && mjs-entry src/index.ts > lib/index.mjs`, it'd still build the lib: https://github.com/telegraf/telegraf/blob/732f2c67bf8f55c9b541bddc45ff0c3e2c593dc3/docs/examples/tsconfig.json#L11

Reply:
I'm very open for restructurings. That said, the idea was the following:
`prepare`: runs the build automatically before `npm publish`
`typedoc`: generates docs

`build`: no-brainer, builds everything from scratch, e.g. after fresh `npm i`
`build:lib`: builds the lib without the doc generation overhead, currently reuses the prepare hook for that
`build:docs`: checks the docs for types and then builds them, currently reuses the doc generation script for the build

Therefore, the difference between between `build:lib` and `prepare` is that `build:lib` is a name for humans to use and `prepare` is a name for npm to use. The other thing is that `build:docs` typechecks the docs before generating them with `typedoc`.

I like to have a naming scheme that makes sense to invoke manually, even if it is just aliases for npm hooks. I don't have a preference in which direction that alias should go, and again, we can also make something totally different from what I suggested here :)

Reply:
Please open a separate pull request. I'll drop these upon rebasing to keep discussion focused and avoid merge conflicts.

Reply:
> `build`: no-brainer, builds everything from scratch, e.g. after fresh `npm i`

`prepare` runs automatically after `npm i`.

Reply:
> Please open a separate pull request.

Confer #1191. Note that it is not possible to include docs typechecks in that PR as it depends on this one.